### PR TITLE
Let FlutterActivity/Fragment/FragmentActivity have an app bundle path override instead of eager resolving during construction

### PR DESCRIPTION
**Pull request by xster**
fixes flutter/flutter#64458

FlutterActivity and FlutterFragment are actually fine as is. Changing them too for consistency. 
FlutterActivity doesn't expose the bundle path via Intent. FlutterFragment has a builder for override only and that doesn't resolve until after the FlutterFragment is created. 

FlutterFragmentActivity tries to eager resolve to populate all the values of the FlutterFragment builder parameters.

### CODE REVIEW

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java [hunk 1]
```diff
@@ -307,6 +310,35 @@ public void itExecutesDartEntrypointProvidedByHost() {
     verify(mockFlutterEngine.getDartExecutor(), times(1)).executeDartEntrypoint(eq(dartEntrypoint));
   }
 
+  @Test
+  public void itUsesDefaultFlutterLoaderAppBundlePathWhenUnspecified() {
+    // ---- Test setup ----
+    FlutterLoader mockFlutterLoader = mock(FlutterLoader.class);
+    when(mockFlutterLoader.findAppBundlePath()).thenReturn("default_flutter_assets/path");
+    FlutterInjector.setInstance(
+        new FlutterInjector.Builder().setFlutterLoader(mockFlutterLoader).build());
```
Comment (xster) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java:319]: that other episode's so I can add this

File: shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java [hunk 2]
```diff
@@ -346,6 +347,10 @@ private void doInitialFlutterViewRun() {
       return;
     }
 
+    if (!FlutterInjector.instance().flutterLoader().initialized()) {
+      throw new AssertionError("Flutter UI can only be started once the Flutter engine's running.");
```
Comment (mehmetf) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java:351]: If users hit this, will they know what to do? Perhaps we can guide them to add-to-app guide or something?

**Comment by ReniDelonzek** [comment]
Hello, has this been published in the dev pub? I am using the latest versions available today and I still have this problem.

For me it only occurs when a notification arrives via firebase messaging and the app is closed

**Comment by ReniDelonzek** [comment]
Based on
[this](https://github.com/transistorsoft/flutter_background_fetch/issues/160) solution, add 

```
<application
        android:name="io.flutter.app.FlutterApplication"
```

solved my problem, However, this should not be necessary after flutter 1.20

**Pull request merged**